From 1df7e4f5c1090631ff6584176f7d1370d08bc15e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <petr.pisar@atlas.cz>
Date: Sat, 16 Apr 2016 13:04:59 +0200
Subject: [PATCH] Include sys/sysmacros.h for major(3)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

GNU major(3) manual page prescribes <sys/types.h>, but that does not work with
sys-libs/glibc-2.23-r1.

https://bugs.gentoo.org/show_bug.cgi?id=580044
Signed-off-by: Petr Písař <petr.pisar@atlas.cz>
---
 config/udev.c                          | 1 +
 hw/xfree86/common/xf86Xinput.c         | 2 ++
 hw/xfree86/os-support/linux/lnx_init.c | 1 +
 3 files changed, 4 insertions(+)

diff --git a/config/udev.c b/config/udev.c
index 28c2658..62908f4 100644
--- a/config/udev.c
+++ b/config/udev.c
@@ -30,6 +30,7 @@
 #include <libudev.h>
 #include <ctype.h>
 #include <unistd.h>
+#include <sys/sysmacros.h>
 
 #include "input.h"
 #include "inputstr.h"
diff --git a/hw/xfree86/common/xf86Xinput.c b/hw/xfree86/common/xf86Xinput.c
index c56a2b9..d33c165 100644
--- a/hw/xfree86/common/xf86Xinput.c
+++ b/hw/xfree86/common/xf86Xinput.c
@@ -86,6 +86,8 @@
 #include <unistd.h>
 #ifdef HAVE_SYS_MKDEV_H
 #include <sys/mkdev.h>          /* for major() & minor() on Solaris */
+#else
+#include <sys/sysmacros.h>
 #endif
 
 #include "mi.h"
diff --git a/hw/xfree86/os-support/linux/lnx_init.c b/hw/xfree86/os-support/linux/lnx_init.c
index 1ed213c..6caf531 100644
--- a/hw/xfree86/os-support/linux/lnx_init.c
+++ b/hw/xfree86/os-support/linux/lnx_init.c
@@ -38,6 +38,7 @@
 #include "xf86_OSlib.h"
 
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 
 #ifndef K_OFF
 #define K_OFF 0x4
-- 
2.8.1

